src/components/SignupPage.jsx

```diff
@@ -0,0 +1,142 @@
+import React, { Component } from 'react'
+import Greetings from './Greetings'
+
+export default class SignupPage extends Component {
+    constructor(props){
+        super(props)
+        this.state = {
+            form: {
+                email: '',
+                password: '',
+                nationality: ''
+            },
+            users: [],
+            validForm: null,
+            errorMessage: false
+
+        }
+        this.initialForm = {...this.state.form}
+    }
+
+    handleChange = ({target}) => {
+
+        const isValid = target.name === 'email' ? 
+        this.checkEmail(target.value) : 
+        this.checkPassword(target.value);
+
+        if (isValid) {
+            this.setState({
+                form: {
+                    ...this.state.form,
+                    [target.name]: target.value},
+                ...this.state.users,
+                validForm: 'is-valid'
+                })    
+        } else {
+            this.setState({
+                form: {...this.state.form,
+                    [target.name]: target.value},
+                users: [...this.state.users],
+                validForm: 'is-invalid'
+            })
+        }
+        
+    }
+
+    checkEmail = (value) => {
+        const regEx = /^\S+@\S+\.\S+$/
+        return value.match(regEx)
+    }
+
+    checkPassword = (value) => {
+        const regEx = /(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,}/
+        return value.match(regEx)
+    }
+
+
+
+    handleSubmit = (e) => {
+        e.preventDefault();
+        if (this.state.validForm === 'is-valid') {
```

Comment:
`this.state.validForm` deberia ser un bool y listo

`if(this.state.validForm)`

por lo general si hay un error le queres avisar al usuario para que lo arregle y no resetear el form

```
if(this.state.validForm) {
  this.setState(state => ({
    ...state,
 form: {...this.initialForm},
 users: [{...state.form}, ...state.users],
}))
```
}